README.md

```diff
@@ -1,17 +1,176 @@
+
+``` ipldsch
+type Principal union {
+  -- Specification defines principals in terms of did:key's (that are multicodec
+  -- identifier for the public key type followed by the raw bytes of the key)
+  -- we represent those as raw public key bytes prefixed with public key multiformat
+  -- code.
+  | secp256k1  "0xe7"
+  | BLS12381G1 "0xea"
+  | BLS12381G2 "0xeb"
+  | Ed25519    "0xed"
+  | P256       "0x1200"
+  | P384       "0x1201"
+  | P512       "0x1202"
+  | RSA        "0x1205"
+  
+  -- To accomodate additional DID methods we represent those as UTF8 encoding
+  -- of the DID omitting `did:` prefix itself. E.g. `did:dns:ucan.xyz` can be
+  -- represented as [0xd1d, ...new TextEncoder().encode('dns:ucan.xyz')] bytes.
+  | DID "0x0d1d"
+} representation bytesprefix
+```
+
+## 2.2 Capability
+
+``` ipldsch
+type Capability struct {
+  with Resource
+  can Ability
+  nb optional NonNormativeFields
+}
+```
+
+### 2.2.1 Resource
+
+The resource pointer MUST be formatted as a [URI](https://www.rfc-editor.org/rfc/rfc3986).
+
+``` ipldsch
+type Resource = String
+```
+
+### 2.2.2 Ability
+
+Abilities MUST be lowercase, and MUST be namespaced with `/` delimeters. Abilities SHOULD have at least one path segment.
+
+``` ipldsch
+type Ability = String
+```
+
+### 2.2.3 `nb` Non-Normative Fields
+
+The `nb` capability field is OPTIONAL. When present, it MUST contain any additional domain specific details and/or restrictions of the capability. 
+
+``` ipldsch
+type NonNormativeFields = { String: Any }
+```
+
+## 2.3 Fact
+
+Facts MUST be structured as maps with string keys.
+
+``` ipldsch
+type Fact { String: Any }
+```
+
+## 2.4 Proof
+
+As of UCAN 0.9, all proofs MUST be given as links (CIDs). Note that UCANs MAY be inlined via the [identity multihash (`0x00`)](https://github.com/multiformats/multicodec/blob/master/table.csv#L2) CID.
+
+``` ipldsch
+&UCAN
+```
+
+## 2.5 Signature
+
+The signature MUST be computed by first encoding it as a [canonical JWT](#3-jwt-canonicalization), and then signed with the issuer's private key. Signatures MUST be encoded as the _varsig_ multiformat representation `<varint sig_alg_code><vairint sig_size><bytes sig_output>`. _Varsig_ multiformat codes SHOULD be [registered multicodec codes](https://github.com/multiformats/multicodec). Unregistered signature types MAY be added via the `NonStandardSignature` prefix.
+
+``` ipldsch
+type Signature union {
+  -- Algorithms here are expected to be valid "varsig" multiformat codes.
+  | NonStandard "0xd000"
+  | ES256K      "0xd0e7" -- secp256k1
+  | BLS12381G1  "0xd0ea" -- NB: G1 is a signature for BLS-12381 G2
+  | BLS12381G2  "0xd0eb" -- NB: G2 is a signature for BLS-12381 G1
+  | EdDSA       "0xd0ed"
+  | ES256       "0xd01200"
+  | ES384       "0xd01201"
+  | ES512       "0xd01202"
+  | RS256       "0xd01205"
+  | EIP191      "0xd191"
```

Comment:
I moved EIP-191 to the "standard" table. I realize that it's not IANA, but it the only way that wallets like MetaMask will sign data. FVM is going to need this as part of the FEVM project (since they want to plug into Ethereum tooling)